akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -668,10 +668,15 @@ import akka.stream.snapshot._
         logicSnapshots(logicIndexes(connection.inOwner)),
         logicSnapshots(logicIndexes(connection.outOwner)),
         connection.portState match {
-          case InReady  => ConnectionSnapshot.ShouldPull
-          case OutReady => ConnectionSnapshot.ShouldPush
-          case x if (x | InClosed | OutClosed) == (InClosed | OutClosed) =>
+          case InReady                                                     => ConnectionSnapshot.ShouldPull
+          case OutReady                                                    => ConnectionSnapshot.ShouldPush
+          case x if (x & (InClosed | OutClosed)) == (InClosed | OutClosed) =>
```

Comment:
I'm not sure this logic is correct: if `x` is `InFailed | OutClosed`, `x & (InClosed | OutClosed)` would be `OutClosed` but `OutClosed` != `(InClosed | OutClosed)`...

Shouldn't the conditional be something like `if (x & (InClosed | OutClosed)) > 0` to match the comment?

Reply:
Though the original code looks even worse: AFAICS `if (x | InClosed | OutClosed) == (InClosed | OutClosed)` means 'neither side is closed'...

Reply:
From the docs up top
```
InFailed:  Always set in conjunction with InClosed. Indicates that the close event is a failure
```

So I think this is safe

Reply:
:+1: on a closer look I agree!